src/app/events/GuildMemberUpdate.ts

```diff
@@ -22,6 +23,10 @@ export default class implements DiscordEvent {
 			return;
 		}
 		
+		if (oldMember.nickname !== newMember.nickname && ServiceUtils.runUsernameSpamFilter(newMember as GuildMember)) {
```

Comment:
const spamCheck = await ServiceUtils.runUsernameSpamFilter(newMember as GuildMember);
oldMember.nickname !== newMember.nicknam && spamCheck

Reply:
I have it in that order to short circuit and not running the spam filter if their nickname did not change.

Reply:
oh that's fine I think it's just missing await. I thought it might be easier to read if it was split across lines but because the order matters here then adding await should work